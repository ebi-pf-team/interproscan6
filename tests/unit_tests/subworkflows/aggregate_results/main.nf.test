nextflow_workflow {

    name "Test Workflow AGGREGATE_RESULTS"
    script "interproscan/subworkflows/aggregate_results/main.nf"
    workflow "AGGREGATE_RESULTS"

    stage {
        symlink "tests/unit_tests/test_inputs/aggregate_results/cdd_matches.json"
        symlink "tests/unit_tests/test_inputs/aggregate_results/hamap_matches.json"
        symlink "tests/unit_tests/test_inputs/aggregate_results/corrupted.json"
    }

    test("Should run successfully") {
        when {
            params {
                 }
            workflow {
                """
                input[0] = Channel.from(file("meta/tests/unit_tests/test_inputs/aggregate_results/cdd_matches.json"), \
                file("meta/tests/unit_tests/test_inputs/aggregate_results/hamap_matches.json"))
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
            assert workflow.exitStatus == 0
        }

    }

    test("Should handle corrupted JSON file error") {
        when {
            params {
            }
            workflow {
                """
                input[0] = Channel.from(file("meta/tests/unit_tests/test_inputs/aggregate_results/corrupted.json"))
                """
            }
        }

        then {
            assert workflow.failed
            assert workflow.exitStatus == 22
        }
    }

    test("Should handle file not found error") {
        when {
            params {
            }
            workflow {
                """
                input[0] = Channel.from(file("meta/tests/unit_tests/test_inputs/aggregate_results/no_hay_banda.json"), file("meta/tests/unit_tests/test_inputs/aggregate_results/cdd_matches.json"))
                """
            }
        }

        then {
            assert workflow.failed
            assert workflow.exitStatus == 5
        }
    }
}
