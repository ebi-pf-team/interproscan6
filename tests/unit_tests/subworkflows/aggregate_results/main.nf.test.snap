{
    "Should pass": {
        "content": [
            {
                "0": [
                    "results_aggregated.json:md5,fb58604aca6dba51bf4acbaa19edb238"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-02T20:40:57.449253"
    },
    "Should run successfully": {
        "content": [
            {
                "0": [
                    "aggregated_result.json:md5,cfb3dd4440fb052bececa882e40f6481"
                ],
                "ch_aggregated_results": [
                    "aggregated_result.json:md5,cfb3dd4440fb052bececa882e40f6481"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-09-23T16:32:33.651976"
    }
}